### PR TITLE
Deprecate v3 in v4

### PR DESCRIPTION
**Pull request by dirkkul**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -23,6 +23,15 @@ The client is tested for python 3.8 and higher.
 
 Visit the official `Weaviate <https://weaviate.io/>`_ website for more information about the Weaviate and how to use it in production.
 
+Client Versions
+--------
+We currently support the following versions client versions:
+
+- 4.X: actively supported
+- 3.X: deprecated, receives only critical bug fixes and dependency updates
+- copy of the 3.X branch in v4 releases: Will be removed at 30.11.2024
```
Comment (daveatweaviate) [inline README.rst:32]: Consider:

 Will be removed on 2024-11-30.

File: weaviate/warnings.py [hunk 2]
```diff
@@ -160,12 +160,14 @@ def root_module_import(name: str, loc: str) -> None:
...
     def weaviate_v3_client_is_deprecated() -> None:
         warnings.warn(
-            message="""Dep016: Python client v3 `weaviate.Client(...)` connections and methods are deprecated. Update
-            your code to use Python client v4 `weaviate.WeaviateClient` connections and methods.
-
-            For Python Client v4 usage, see: https://weaviate.io/developers/weaviate/client-libraries/python
-            For code migration, see: https://weaviate.io/developers/weaviate/client-libraries/python/v3_v4_migration
-            """,
+            message="""Dep016: Python client v3 `weaviate.Client(...)` connections and methods are deprecated and will
+            be removed by 30.11.2024.
```
Comment (daveatweaviate) [inline weaviate/warnings.py:164]: Consider

removed by 2024-11-30.

File: weaviate/warnings.py [hunk 3]
```diff
@@ -160,12 +160,14 @@ def root_module_import(name: str, loc: str) -> None:
...
+
+            You can either:
+            a) upgrade your code to use Python client v4 `weaviate.WeaviateClient` connections and methods.
```
Comment (daveatweaviate) [inline weaviate/warnings.py:167]: Consider:

```
(Drop you can either)

Upgrade your code to use Python client v4 `weaviate.WeaviateClient` connections and methods.
    - For Python Client v4 usage, see: https://weaviate.io/developers/weaviate/client-libraries/python
     - For code migration, see: https://weaviate.io/developers/weaviate/client-libraries/python/v3_v4_migration
            
If you have to use v3 code, install the v3 client. Pin the v3 dependency in your requirements file: `weaviate-client>=3.26.6;<4.0.0`""",

```

**Review by daveatweaviate — COMMENTED** [review]
@dirkkul  thanks for the update, I left some suggestions on the wording

**Review by tsmith023 — APPROVED** [review]

**Pull request merged**